### PR TITLE
fix(ts): add types for smart sort

### PR DESCRIPTION
**Pull request by eunjae-lee**
## Summary

This PR adds the following types for smart sort

- `relevancyStrictness` in the search parameter
- `nbSortedHits` and `appliedRelevancyStrictness` in the search response

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1156,6 +1166,14 @@ declare namespace algoliasearchHelper {
      * total number of hits of this query on the index
      */
     nbHits: number;
+    /**
+     * total number of hits from smart sort
+     */
+    nbSortedHits?: number;
```
Comment (shortcuts) [inline index.d.ts:1172]: We should state the word `smart` to avoid confusion with any kind of other sort, like `nbSmartSortedHits`, wdyt?

Comment (eunjae-lee) [reply index.d.ts:1172]: That's a property sent by the engine, so we cannot change it :)

File: index.d.ts [hunk 2]
```diff
@@ -1156,6 +1166,14 @@ declare namespace algoliasearchHelper {
      * total number of hits of this query on the index
      */
     nbHits: number;
+    /**
+     * total number of hits from smart sort
```
Comment (Haroenv) [inline index.d.ts:1170]: can we use the same descriptions as in the client? https://github.com/algolia/algoliasearch-client-javascript/pull/1251

Comment (eunjae-lee) [reply index.d.ts]: Oh nice, I didn't know they were already there. https://github.com/algolia/algoliasearch-helper-js/pull/809/commits/107374a11b059135098734586cef082742bd256c

File: index.d.ts [hunk 3]
```diff
@@ -511,6 +511,13 @@ declare namespace algoliasearchHelper {
     ruleContexts?: string[];
     optionalFilters?: Array<string | string[]>;
     queryLanguages?: string[];
+
+    /**
+     * The relevancy threshold to apply to search in a virtual index [0-100]. A Bigger
+     * value means fewer, but more relevant results, smaller value means more, but
+     * less relevant results.
+     */
+    relevancyStrictness?: number;
```
Comment (Haroenv) [inline index.d.ts:520]: this one likely isn't needed

Comment (eunjae-lee) [reply index.d.ts:520]: Can you tell me why it's not needed?
When we need to set `relevancyStrictness` from the frontend side, we will need it, no?

Comment (Haroenv) [reply index.d.ts:520]: because the interface inherits from the client type, but in this case we still want to copy it for v3 compact

**Review by Haroenv — APPROVED** [review]

**Review by shortcuts — APPROVED** [review]

File: index.d.ts (hunk 3 above)
Comment (eunjae-lee) [reply index.d.ts:520]: Ah got what you mean 👍🏼

**Pull request merged**